src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Serialization.cs

```diff
@@ -153,6 +152,35 @@ public void WriteTo(ObjectWriter writer)
                     writer.WriteInt32(v);
                 }
             }
+
+            if (_simpleTypeNameToExtensionMethodMap == null)
```

Comment:
is there a reason yuo need this to be null, instead of just a singleton empty instance?  that would make reading/writing/accessing simpler.

Reply:
comment still applies.

Reply:
it's mutable

Reply:
let me rephrase.  instead of the field being null, just have the field be a separate empty dictionary.  that way you don't need all the null checks everywhere.  nbd though.